arrows/qt/EmbeddedPipelineWorker.cxx

```diff
@@ -49,6 +51,34 @@ class EmbeddedPipeline : public kwiver::embedded_pipeline
   bool outputConnected_ = false;
 };
 
+// ----------------------------------------------------------------------------
+EmbeddedPipeline
+::EmbeddedPipeline( RequiredEndcaps endcaps )
+  : endcaps_{ endcaps }
+{
+  auto ss = []( QString const& qs ){
+    auto const& data = qs.toLocal8Bit();
+    auto const l = static_cast< size_t >( data.size() );
+    return std::string{ data.constData(), l };
+  };
+
+  auto const& an = ss( qApp->applicationName() );
+  auto const& av = ss( qApp->applicationVersion() );
+  auto ap = std::string{};
+
+  auto const appPath = qApp->applicationDirPath();
+  if( appPath.endsWith( QStringLiteral( "bin" ) ) )
```

Comment:
Do we have use cases where the application is not installed in `bin`?  I'm not sure that we handle this case elsewhere when looking for config files relative to the executable.

Reply:
Where is "elsewhere"?

In other cases, the caller of the config API's has been much more closely coupled to the application, which generally knows if it is in `<prefix>/bin` or not. Here, however, since we don't know who is using this, we cannot make such assumptions, and since this is library code, it should strive to be as permissive as possible, within reason. Some Windows (and possibly macOS) applications may have the application binary in the root of the install prefix.

The alternative would be to require the user of `EmbeddedPipelineWorker` to also set the application information. (Or to provide it optionally, but that API could be added later, as/if needed.)

Reply:
I'll have to look. I thought there were other places in KWIVER where we used `[executable path]/..` to find things.  These would have used KWSys and not Qt to get the executable path.

Reply:
You're probably thinking of [here](https://github.com/Kitware/kwiver/blob/8b55a9c32aabb4b895f95d01666209d316a877d9/vital/applets/kwiver_applet.cxx#L33) :slightly_smiling_face:. (There are other uses, but that one is the most direct parallel.) Note that, in that case, because we are dealing with a known executable under our control (not the case for `EmbeddedPipelineWorker`), we can safely¹ assume that the application is in fact in `<prefix>/bin`.

(¹ For installations, anyway; I think this might be incorrect for build trees using multi-config generators.)

Relatedly, [whereami.c](https://github.com/Kitware/kwiver/blob/master/vital/util/whereami.c) was created specifically to provide the functionality of [QCoreApplication::applicationDirPath](https://doc.qt.io/qt-5/qcoreapplication.html#applicationDirPath) for KWIVER, which of course doesn't depend on Qt.

Reply:
Yes, that's what I was thinking of, and there might be a couple more too, but your point is taken.  In KWIVER we always know where we are installing our binaries.

That said, adding this sort of check elsewhere couldn't hurt.  We might consider making a common function for this in `get_paths.h`.  Maybe `get_executable_root_path()` or similar name.  We may need to add additional checks in the future, like walking up the directory structure until we find `lib` or `share` to help with the `Release/bin` case.  It would be nice if we wrote this code once and used it in multiple places.  There are many places we use `applicationDirPath()` or `get_excutable_path()` in KWIVER (and in apps like TeleSculptor) and then append `/..` because we really are looking for the root containing `lib` or `share`.

Reply:
Maybe, but is that likely enough that it needs to be part of this PR? Since we're anyway using `QCoreApplication` for the other application information, it's consistent and simple to get the prefix as-is.

Looking for `lib` won't work; a non-broken install on 64-bit Fedora will probably not have `lib` (there's a small chance a 64-bit Fedora install won't even have `/usr/lib`). Looking for `share` is probably best, particularly since we make basically zero effort to support alternate names to `share`. However, I don't think this is actually needed in most cases other than non-installed builds with multi-config generators. (And TBH, I mainly care about supporting installs.)